### PR TITLE
Render out taxonomy items to json for react filters

### PR DESCRIPTION
**Pull request by kevinhowbrook**
Renders out taxonomy values to the template as json

### CODE REVIEW

File: rca/project_styleguide/templates/patterns/pages/programmes/programme_index.html [hunk 1]
```diff
@@ -20,6 +20,17 @@ <h1 class="title-area__header heading heading--display-two">{{ title_multiline|l
...
+            <div class="filters">
+                {% for f in filters %}
+                    <ul class="filters filters__list">
+                        {{ f.title }}
+                        {% for i in f.items %}
+                            <li class="filters filters__list item">{{ i }}</li>
+                        {% endfor %}
+                    </ul>
+                {% endfor %}
+            </div>
```
Comment (kevinhowbrook) [inline rca/project_styleguide/templates/patterns/pages/programmes/programme_index.html]: @thibaudcolas feel free to remove this markup, just there for testing purposes really and as an example

File: rca/programmes/models.py [hunk 2]
```diff
@@ -707,6 +707,20 @@ def get_context(self, request, *args, **kwargs):
         except EmptyPage:
             subpages = paginator.page(paginator.num_pages)
 
+        # Listing filters
+        degree_levels = [i.title for i in DegreeLevel.objects.all()]
+        degree_levels_title = DegreeLevel._meta.verbose_name.capitalize()
```
Comment (thibaudcolas) [inline rca/programmes/models.py]: Oh that’s pretty cool, didn’t think we could do that!

File: rca/programmes/models.py [hunk 3]
```diff
@@ -707,6 +707,20 @@ def get_context(self, request, *args, **kwargs):
         except EmptyPage:
             subpages = paginator.page(paginator.num_pages)
 
+        # Listing filters
+        degree_levels = [i.title for i in DegreeLevel.objects.all()]
```
Comment (thibaudcolas) [inline rca/programmes/models.py]: I would also need the key/id for each item so they can be passed over to the API to apply the correct filter.

File: rca/programmes/models.py [hunk 4]
```diff
@@ -707,6 +707,20 @@ def get_context(self, request, *args, **kwargs):
...
+        degree_levels_title = DegreeLevel._meta.verbose_name.capitalize()
+        programme_types = [i.display_name for i in ProgrammeType.objects.all()]
+        programme_types_title = ProgrammeType._meta.verbose_name.capitalize()
+        # TODO schools and subject
+
+        filters = [
+            {"title": degree_levels_title, "items": degree_levels},
+            {"title": programme_types_title, "items": programme_types},
+            {"title": "Subject", "items": ["T", "B", "D"]},
```
Comment (thibaudcolas) [inline rca/programmes/models.py]: 😄

File: rca/project_styleguide/templates/patterns/pages/programmes/programme_index.html (hunk 1 above)
Comment (thibaudcolas) [reply rca/project_styleguide/templates/patterns/pages/programmes/programme_index.html]: 👍 good idea

File: rca/programmes/models.py (hunk 2 above)
Comment (kevinhowbrook) [reply rca/programmes/models.py]: Yea - it's unlikely to change, but if it does, at least we only change it in one place

File: rca/programmes/models.py (hunk 3 above)
Comment (kevinhowbrook) [reply rca/programmes/models.py]: added iID in https://github.com/torchbox/rca-wagtail-2019/pull/274/commits/477264c4c7826f47a1b393ef9f6577bedce46035

**Pull request merged**